### PR TITLE
Add Network and account API

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds access to getNameInfo method, and all other methods listed in blockstack.js for network and transactions.
* adds network propery to BlockstackSession for the current network config
* updates blockstack.js to commit https://github.com/blockstack/blockstack.js/commit/211455e10b86d298e58e607367d7f4dd2583e709
* adds a button in example to get name info of dev_android_sdk.id.blockstack
* adds tests for getNameInfo

To test
1. checkout branch 'feature/name_info`
1. start example in Android Studio
1. see new button at very bottom

### CODE REVIEW

**Comment by friedger** [comment]
We want to have a better network api

File: blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt [hunk 1]
```diff
@@ -65,10 +68,18 @@ class BlockstackSession(context: Context? = null, private val config: Blockstack
     private var getAppBucketUrlCallback: ((Result<String>) -> Unit)? = null
     private var getUserAppFileUrlCallback: ((Result<String>) -> Unit)? = null
 
-
-    private val v8blockstackAndroid: V8Object
+    internal val v8blockstackAndroid: V8Object
     private val v8userSessionAndroid: V8Object
+    private val v8networkAndroid: V8Object
     private val v8userSession: V8Object
```
Comment (shreyasthiagaraj) [inline blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt]: This should be camelCased, and is "Android" necessary in the name?
Also, would it make sense to have the object type at the end of the name?

i.e.
blockstackV8
networkV8

What is the difference between v8UserSession and v8UserSessionAndroid?

File: blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt [hunk 2]
```diff
@@ -629,12 +646,21 @@ class BlockstackSession(context: Context? = null, private val config: Blockstack
     }
 }
 
+/**
+ * Executor defines where functions are executed. Three different cases are distinguished:
+ * 1. main thread: to start an intent launching the login process. This has to be the UI thread
+ * 1. network thread: to make network calls. This must not be the UI thread, it is usually a thread from CommonPool
```
Comment (shreyasthiagaraj) [inline blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt]: Minor nit: Add period at the end of these comments, just to be consistent with full sentences :).

File: blockstack-sdk/src/main/java/org/blockstack/android/sdk/Network.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.eclipsesource.v8.V8
+import com.eclipsesource.v8.V8Array
+import com.eclipsesource.v8.V8Object
+import org.json.JSONObject
+
+/**
+ * Object giving access to information about the blockstack network.
+ */
+class Network internal constructor(private val v8networkAndroid: V8Object, val v8: V8) {
```
Comment (shreyasthiagaraj) [inline blockstack-sdk/src/main/java/org/blockstack/android/sdk/Network.kt]: What was the thinking behind not just placing these methods in the BlockstackSession object, i.e. `BlockstackSession().getNameInfo(...)`?

File: blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt (hunk 1 above)
Comment (friedger) [reply blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt]: The names reflect the location in the javascript world. The prefix indicate that they are v8 object and have a correspondence in there. Then name that follows is the same in javascript. So
v8networkAndroid in Android is the representation of networkAndroid object in javascript.

All the *Android javascript objects are defined in blockstack_android.js. 

userSession is the blockstack object that can be called directly if the method signatures are transferrable between the two worlds, userSessionAndroid is the bridge object that translates `Promise`s and the like into String, etc. These *Android object make also calls to the java world.

File: blockstack-sdk/src/main/java/org/blockstack/android/sdk/Network.kt (hunk 3 above)
Comment (friedger) [reply blockstack-sdk/src/main/java/org/blockstack/android/sdk/Network.kt]: BlockstackSession is the object that represents a app session. All the transaction methods do not belong to an app/domain. Therefore it is better to have them separated. In the end, the API should provide a builder that does not need a `BlockstackSession` at all if the developer wants to configure the network themselves.

Comment (shreyasthiagaraj) [reply blockstack-sdk/src/main/java/org/blockstack/android/sdk/Network.kt]: Got it.

File: blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt (hunk 1 above)
Comment (shreyasthiagaraj) [reply blockstack-sdk/src/main/java/org/blockstack/android/sdk/BlockstackSession.kt]: Thanks for explaining.

**Comment by shreyasthiagaraj** [comment]
@friedger This looks good. The ones in the docs should be sufficient. Please add documentation and I'll merge this in--thank you! 👍

**Pull request merged**